__init__.py

```diff
@@ -27,41 +27,141 @@
+    location_format
+
 import re
+from lingua_franca.format import nice_duration
 
+from datetime import datetime
+import pytz
 
 
-class DirectorySkill(NeonSkill):
+class DirectorySkill(KioskSkill):
 
     def __init__(self):
         super(DirectorySkill, self).__init__(name="DirectorySkill")
         self.url = "https://www.alamoanacenter.com/en/directory/"
+        self._speak_timeout = 60
 
+    @property
+    def request_lang(self):
+        return self.lang.split('-')[0]
 
-    def initialize(self):
-        # When first run or prompt not dismissed, wait for load and prompt user
-        if self.settings.get('prompt_on_start'):
-            self.bus.once('mycroft.ready', self._start_mall_parser_prompt)
-
-    @intent_file_handler("run_mall_parser.intent")
-    def start_mall_parser_intent(self, message):
-        LOG.info(message.data)
-        self._start_mall_parser_prompt(message)
-        return
-
-    # @property
+    @property
     def mall_link(self):
         mall_link = 'https://www.alamoanacenter.com/'
         return self.settings.get("mall_link") or mall_link
 
-    def user_request_handling(self, message):
+    @property
+    def timeout_seconds(self) -> int:
+        """
+        Time in seconds to wait for a user response before timing out
+        """
+        return 60
+
+    @property
+    def greeting_dialog(self) -> str:
+        """
+        Specify a dialog to speak on interaction start
+        """
+        return 'greeting'
+
+    @property
+    def goodbye_dialog(self) -> str:
+        """
+        Specify a dialog to speak when an interaction ends cleanly
+        """
+        return 'goodbye'
+
+    @property
+    def timeout_dialog(self) -> str:
+        """
+        Specify a dialog to speak when an interaction ends after some timeout
+        """
+        return 'timeout'
+
+    @property
+    def error_dialog(self) -> str:
+        """
+        Specify a dialog to speak on unhandled errors
+        """
+        return 'unexpected_error'
+
+    def setup_new_interaction(self, message) -> bool:
+        """
+        Override to include skill-specific actions on first user interaction.
+        This is the first action that could prompt user to input language, etc.
+        :param message: Message associated with start request
+        :returns: True if user interaction is supported
+        """
+        if message:
+            return True
+
+    def handle_new_interaction(self, message):
+        """
+        Override to interact with the user after the greeting message has been
+        spoken.
+        :param message: Message associated with start request
+        """
+        self.speak_dialog('how_can_i_help_you')
+
+    def handle_user_utterance(self, message):
+        """
+        Handle any input from a user interacting with the kiosk.
+        :param message: Message associated with user utterance
+        """
+        answer = message.data['utterances']
```

Comment:
I think this should be `message.data['utterances'][0]` so you have a string instead of a list of strings